app/db.py

```diff
@@ -75,57 +84,63 @@ def n_edges(self):
     def node(self, uri):
         # looks up a node by uri (essentially [[wikilink]]).
         # this used to be even worse :)
-        try:
-            node = self.nodes()[uri.lower()]
-            return node
-        except (KeyError, IndexError):
-            # We'll handle 404 in the template, as we want to show backlinks to non-existent nodes.
-            # Return an empty.
-            return Node(uri)
+        # nodes = self.nodes()
+        # try:
+        #     node = nodes[uri.lower()]
+        #     return node
+        # except (KeyError, IndexError):
+        #     # We'll handle 404 in the template, as we want to show backlinks to non-existent nodes.
+        #     # Return an empty.
+        #     return Node(uri)
+        node = query_node(uri.lower())
+        return node
 
     def existing_permutations(self, uri, max_length=4):
         # looks up nodes matching a permutation of the tokenized uri.
-        # 
+        #
         # example use: if [[server-agora]] does not exist, serve [[agora-server]]
         #
         # this is, of course, a terrible implementation and dangerous :)
         # only enable up to 4 tokens as 24 permutations is manageable.
         tokens = uri.split('-')
         permutations = itertools.permutations(tokens, max_length)
         permutations = ['-'.join(permutation) for permutation in permutations]
-        nodes = [node for node in G.nodes(only_canonical=True).values() if node.wikilink in permutations and node.subnodes]
+        nodes = [node for node in G.nodes(only_canonical=True).values(
+        ) if node.wikilink in permutations and node.subnodes]
         return nodes
 
     def match(self, regex):
         # returns a list of nodes reasonably matching a regex.
         current_app.logger.debug(f'*** Looking for nodes matching {regex}.')
-        nodes = [node for node in G.nodes(only_canonical=True).values() if 
-                    # has some content
-                    node.subnodes and
-                    # its wikilink matches the regex
-                    re.match(regex, node.wikilink)
-                ]
+        nodes = [node for node in G.nodes(only_canonical=True).values() if
+                 # has some content
+                 node.subnodes and
+                 # its wikilink matches the regex
+                 re.match(regex, node.wikilink)
+                 ]
         current_app.logger.debug(f'*** Found related nodes: {nodes}.')
         return nodes
 
     def search(self, regex):
         # returns a list of nodes reasonably freely matching a regex.
-        current_app.logger.debug(f'*** Looking for nodes matching {regex} freely.')
-        nodes = [node for node in G.nodes(only_canonical=True).values() if 
-                    # has some content
-                    node.subnodes and
-                    # its wikilink matches the regex
-                    re.search(regex, node.wikilink)
-                ]
+        current_app.logger.debug(
```

Comment:
What's up with these? Is your editor enforcing 80 characters? Can you tell it not to do that? :) It makes the diff harder to read. Reformatting of existing code should ideally be done in a different PR.

Reply:
this isn't on purpose. blame vscode

Reply:
👍